src/aktualizr_primary/secondary.cc

```diff
@@ -82,8 +81,10 @@ class SecondaryWaiter {
     timer_.async_wait([&](const boost::system::error_code& error_code) {
       if (!!error_code) {
         LOG_ERROR << "Wait for secondaries has failed: " << error_code;
+        throw std::runtime_error("Error while waiting for secondary");
```

Comment:
As an alternative to these two 'throws' just one throw can be added at the end of createIPSecondaries().